### PR TITLE
Suppress 'No server certificate CA names sent' message

### LINKED ISSUE #9080: "No server certificate CA names sent"  message is confusing
<pre>
speed% openssl version
OpenSSL 1.1.1b  26 Feb 2019
</pre>

I ran openssl server as follows:

<pre>
speed% openssl s_server -port 29999  -CAfile ca.crt -verify 1 -cert server.pem         
</pre>

I then attempted to connect to this server from my client. The server produced the following output in its terminal:

<pre>
verify depth is 1
Using default temp DH parameters
ACCEPT
-----BEGIN SSL SESSION PARAMETERS-----
MH4CAQECAgMEBAITAgQgGg71gUjGt2LGFJrCZZVsZpRtIEG4KAX/jtXDHu3X6J0E
MO61KEPzOh/yfnWJlsSde8DE6L9PETpav7n1BoStU6TYVZArmMKye/WOfvE2fByr
2KEGAgRc9u4wogQCAhwgpAYEBAEAAACuBwIFAIPYu9Q=
-----END SSL SESSION PARAMETERS-----
Shared ciphers:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES256-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:DHE-RSA-AES128-SHA:AES256-GCM-SHA384:AES128-GCM-SHA256:AES256-SHA256:AES128-SHA256:AES256-SHA:AES128-SHA
Signature Algorithms: ECDSA+SHA256:ECDSA+SHA384:ECDSA+SHA512:Ed25519:Ed448:RSA-PSS+SHA256:RSA-PSS+SHA384:RSA-PSS+SHA512:RSA-PSS+SHA256:RSA-PSS+SHA384:RSA-PSS+SHA512:RSA+SHA256:RSA+SHA384:RSA+SHA512:ECDSA+SHA224:RSA+SHA224:DSA+SHA224:DSA+SHA256:DSA+SHA384:DSA+SHA512
Shared Signature Algorithms: ECDSA+SHA256:ECDSA+SHA384:ECDSA+SHA512:Ed25519:Ed448:RSA-PSS+SHA256:RSA-PSS+SHA384:RSA-PSS+SHA512:RSA-PSS+SHA256:RSA-PSS+SHA384:RSA-PSS+SHA512:RSA+SHA256:RSA+SHA384:RSA+SHA512:ECDSA+SHA224:RSA+SHA224
Supported Elliptic Groups: X25519:P-256:X448:P-521:P-384
Shared Elliptic groups: X25519:P-256:X448:P-521:P-384
---
No server certificate CA names sent
CIPHER is TLS_AES_256_GCM_SHA384
Secure Renegotiation IS supported
ERROR
shutting down SSL
CONNECTION CLOSED
</pre>

This issue is with respect to **No server certificate CA names sent** message. What does it mean? Searching the internet for this message seems to yield as a general description of what happened, "something related to openssl was not correctly configured".

- Is this an error or a status message? Since the server produced it, is this a reference to the client not supplying some important information, or a note that the server is not going to send certificate CA names to the client?
- According to the internet this seems like an error message. But it is produced by the server side (`openssl s_server`) and it is complaining about a server certificate - is this the server complaining about itself?
- Was a certificate not provided, or was a certificate provided but was missing a name? Which name? Common name? OU? Issuer?
- How can a user diagnose and resolve the condition that led to this message/error being printed, assuming this indicates an error?

### PR DESCRIPTION
**Pull request by wbrawner**
Fixes #9080

Signed-off-by: Billy Brawner <billy@wbrawner.com>

### CODE REVIEW

**Comment by mattcaswell** [comment]
Close/reopen to kick CLA bot. There was an error in the CLA database for your email address. Should be fixed now.

File: apps/s_cb.c [hunk 1]
```diff
@@ -1507,7 +1507,7 @@ void print_ca_names(BIO *bio, SSL *s)
     const STACK_OF(X509_NAME) *sk = SSL_get0_peer_CA_list(s);
     int i;
 
-    if (sk == NULL || sk_X509_NAME_num(sk) == 0) {
+    if (!SSL_is_server(s) && (sk == NULL || sk_X509_NAME_num(sk) == 0)) {
```
Comment (mattcaswell) [inline apps/s_cb.c]: I think this is not quite right. Before we were getting the message "No server certificate CA names sent":

````
No server certificate CA names sent
CIPHER is TLS_AES_256_GCM_SHA384
Secure Renegotiation IS supported
````

Now instead, with this patch, we get "Acceptable server certificate CA names":

````
Acceptable server certificate CA names
CIPHER is TLS_AES_256_GCM_SHA384
Secure Renegotiation IS supported
````

Really we want neither of these in the event that the list is empty.

**Comment by mattcaswell** [comment]
> Should that be printed still or is it fine to have it removed?

I think it's fine to remove it.

**Comment by t8m** [comment]
Merged to master and 1.1.1

**Pull request closed without merging**